com.unity.render-pipelines.high-definition/Editor/Material/Decal/DecalProjectorEditor.cs

```diff
@@ -248,110 +338,133 @@ void DrawHandles()
                         {
                             PrefabUtility.RecordPrefabInstancePropertyModifications(decalProjector);
                         }
-                    }
 
-                    // Automatically recenter our transform component if necessary.
-                    // In order to correctly handle world-space snapping, we only perform this recentering when the user is no longer interacting with the gizmo.
-                    if ((GUIUtility.hotControl == 0) && (decalProjector.offset != Vector3.zero))
+                        // Smoothly update the decal image projected
+                        DecalSystem.instance.UpdateCachedData(decalProjector.Handle, decalProjector.GetCachedDecalData());
+                    }
+                }
+            }
+            else if (editMode == k_EditUVAndPivot)
+            {
+                // Pivot
+                using (new Handles.DrawingScope(fullColor, Matrix4x4.TRS(Vector3.zero, decalProjector.transform.rotation, Vector3.one)))
+                {
+                    EditorGUI.BeginChangeCheck();
+                    Vector3 newPosition = ProjectedTransform.DrawHandles(decalProjector.transform.position, .5f * decalProjector.size.z - decalProjector.offset.z, decalProjector.transform.rotation);
+                    if (EditorGUI.EndChangeCheck())
                     {
-                        needToRefreshDecalProjector = true;
+                        Undo.RecordObjects(new UnityEngine.Object[] { decalProjector, decalProjector.transform }, "Decal Projector Change");
 
-                        // Both the DecalProjectorComponent, and the transform will be modified.
-                        // The undo system will automatically group all RecordObject() calls here into a single action.
-                        Undo.RecordObject(decalProjector.transform, "Decal Projector Change");
+                        decalProjector.offset += Quaternion.Inverse(decalProjector.transform.rotation) * (decalProjector.transform.position - newPosition);
+                        decalProjector.transform.position = newPosition;
+                    }
+                }
```

Comment:
Instead of `//Pivot`, I think it will be clearer if you add it's own method. this also applies to `//UV`